hybrid/pkg/agent/agent.go

```diff
@@ -0,0 +1,169 @@
+	initStatus  error
+}
+
+func New() *HybridPluginAgent {
+	interceptor := new(HybridPluginAgentInterceptor)
+	return &HybridPluginAgent{interceptor: interceptor}
+}
+
+func (p *HybridPluginAgent) SetLogger(logger hclog.Logger) {
+	p.logger = logger
+
+}
+func (p *HybridPluginAgent) AidAttestation(stream nodeattestorv1.NodeAttestor_AidAttestationServer) error {
+	if len(p.pluginList) == 0 || p.initStatus != nil {
+		return status.Errorf(codes.InvalidArgument, "Plugin initialization error")
+	}
+
+	p.interceptor.setCustomStream(stream)
+	p.interceptor.SetContext(stream.Context())
+	p.interceptor.SetLogger(p.logger)
+
+	interceptors := []AgentInterceptorInterface{}
+
+	for i := 0; i < len(p.pluginList); i++ {
+		var newInterceptor AgentInterceptorInterface = p.interceptor.SpawnInterceptor()
+		newInterceptor.SetPluginName(p.pluginList[i].PluginName)
+		interceptors = append(interceptors, newInterceptor)
+
+		elem := reflect.ValueOf(p.pluginList[i].Plugin)
+		result := elem.MethodByName("AidAttestation").Call([]reflect.Value{reflect.ValueOf(newInterceptor)})
+		err := result[0].Interface()
+		if err != nil {
+			return status.Errorf(codes.Internal, "An error ocurred when during AidAttestation.")
+		}
+	}
+
+	combinedMessage := common.PluginMessageList{}
+	for i := 0; i < len(interceptors); i++ {
+		combinedMessage.Messages = append(combinedMessage.Messages, interceptors[i].GetMessage())
+	}
+
+	return p.interceptor.SendCombined(combinedMessage)
+}
+
+func (p *HybridPluginAgent) Configure(ctx context.Context, req *configv1.ConfigureRequest) (*configv1.ConfigureResponse, error) {
+	pluginData, _ := p.decodeStringAndTransformToAstNode(req.HclConfiguration)
+
+	pluginNames, pluginsData := p.parseReceivedData(pluginData)
+
+	p.pluginList, p.initStatus = p.initPlugins(pluginNames)
+
+	if len(p.pluginList) == 0 || p.initStatus != nil {
+		return nil, p.initStatus
+	}
+
+	for i := 0; i < len(p.pluginList); i++ {
+		elem := reflect.ValueOf(p.pluginList[i].Plugin)
+		req.HclConfiguration = pluginsData[p.pluginList[i].PluginName]
+
+		result := elem.MethodByName("Configure").Call([]reflect.Value{reflect.ValueOf(ctx), reflect.ValueOf(req)})
+		err := result[1]
+
+		if !err.IsNil() {
+			return &configv1.ConfigureResponse{}, status.Errorf(codes.Internal, "Error configuring one of the supplied plugins.")
+		}
+	}
+
+	return &configv1.ConfigureResponse{}, nil
+}
+
+func (p *HybridPluginAgent) decodeStringAndTransformToAstNode(hclData string) (common.Generics, error) {
+	var genericData common.GenericPluginSuper
+	if err := hcl.Decode(&genericData, hclData); err != nil {
+		return nil, err
+	}
+
+	var data bytes.Buffer
+	printer.DefaultConfig.Fprint(&data, genericData.Plugins)
+
+	var astNodeData common.Generics
+
+	if err := hcl.Decode(&astNodeData, data.String()); err != nil {
+		return nil, err
+	}
+
+	return astNodeData, nil
+}
+
+func (p *HybridPluginAgent) parseReceivedData(data common.Generics) ([]string, map[string]string) {
```

Comment:
in this case it would make sense to use a meaningful return name
e.g
```suggestion
func (p *HybridPluginAgent) parseReceivedData(data common.Generics) (plugins []string, meaningfulName map[string]string) {
```